api/securitytest/gitleaks.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2019 Globo.com authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package securitytest
+
+import (
+	"encoding/json"
+	"strings"
+
+	"github.com/globocom/huskyCI/api/log"
+	"github.com/globocom/huskyCI/api/types"
+)
+
+// GitleaksOutput is the struct that holds all data from Gitleaks output.
+type GitleaksOutput []GitLeaksIssue
+
+// GitLeaksIssue is the struct that holds all isssues from Gitleaks output.
+type GitLeaksIssue struct {
+	Line          string `json:"line"`
+	Commit        string `json:"commit"`
+	Offender      string `json:"offender"`
+	Rule          string `json:"rule"`
+	Info          string `json:"info"`
+	CommitMessage string `json:"commitMsg"`
+	Author        string `json:"author"`
+	Email         string `json:"email"`
+	File          string `json:"file"`
+	Repository    string `json:"repo"`
+	Date          string `json:"date"`
+	Tags          string `json:"tags"`
+	Severity      string `json:"severity"`
+}
+
+func analyseGitleaks(gitleaksScan *SecTestScanInfo) error {
+	gitLeaksOutput := GitleaksOutput{}
+	gitleaksScan.FinalOutput = gitLeaksOutput
+
+	// nil cOutput states that no Issues were found.
+	if gitleaksScan.Container.COutput == "" {
+		gitleaksScan.prepareContainerAfterScan()
+		return nil
+	}
+
+	// Unmarshall rawOutput into finalOutput, that is a GitleaksOutput struct.
+	if err := json.Unmarshal([]byte(gitleaksScan.Container.COutput), &gitLeaksOutput); err != nil {
+		log.Error("analyzeGitleaks", "GITLEAKS", 1038, gitleaksScan.Container.COutput, err)
+		gitleaksScan.ErrorFound = err
+		gitleaksScan.prepareContainerAfterScan()
+		return err
+	}
+	gitleaksScan.FinalOutput = gitLeaksOutput
+
+	// check results and prepare all vulnerabilities found
+	gitleaksScan.prepareGitleaksVulns()
+	gitleaksScan.prepareContainerAfterScan()
+	return nil
+}
+
+func (gitleaksScan *SecTestScanInfo) prepareGitleaksVulns() {
+
+	huskyCIgitleaksResults := types.HuskyCISecurityTestOutput{}
+	gitleaksOutput := gitleaksScan.FinalOutput.(GitleaksOutput)
+	for _, issue := range gitleaksOutput {
+		// golang: vendor issues should not effect us
+		if strings.HasPrefix(issue.File, "vendor/") && strings.HasSuffix(issue.File, ".go") {
+			continue
+		}
```

Comment:
Hey, @abzcoding thanks for your contribution!

@rafaveira3 should we ignore vendor issues?
@abzcoding and if we ignore vendor files, why only `.go` files?

Reply:
Hey, @joserenatosilva! Good catch! I think we can indeed consider vendor issues as they may impact the security of the repository being analyzed by huskyCI. @abzcoding, would you mind removing this if condition? 🙃

Reply:
the reason to ignore them is that they're not our sensitive data :), it's somebody else's concern ;)

@joserenatosilva  you're right if we decide to ignore vendor issues, then we don't need to only check `.go` files

Reply:
@joserenatosilva  @rafaveira3 should I remove the condition or not, I'm bit confused now 😕

Reply:
Sorry about that! Yep, let's remove the if condition and also the comment right above it. 😃

Reply:
ok then, thank you for clarifying the situation, but bear in mind that it would generate a lot of issues 😈